File: web/src/plugin/inspection/api/template.js
import service from '@/utils/request'

export const createTemplate = (data) => {
  return service({ url: '/inspectionTemplate/createTemplate', method: 'post', data })
}

export const deleteTemplate = (params) => {
  return service({ url: '/inspectionTemplate/deleteTemplate', method: 'delete', params })
}

export const updateTemplate = (data) => {
  return service({ url: '/inspectionTemplate/updateTemplate', method: 'put', data })
}

export const findTemplate = (params) => {
  return service({ url: '/inspectionTemplate/findTemplate', method: 'get', params })
}

export const getTemplateList = (params) => {
  return service({ url: '/inspectionTemplate/getTemplateList', method: 'get', params })
}

File: server/plugin/inspection/router/template.go
package router

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

var Template = new(templateRouter)

type templateRouter struct{}

func (r *templateRouter) Init(private *gin.RouterGroup) {
	{
		group := private.Group("inspectionTemplate").Use(middleware.OperationRecord())
		group.POST("createTemplate", apiTemplate.CreateTemplate)
		group.POST("copyTemplate", apiTemplate.CopyTemplate)
		group.DELETE("deleteTemplate", apiTemplate.DeleteTemplate)
		group.PUT("updateTemplate", apiTemplate.UpdateTemplate)
	}
	{
		group := private.Group("inspectionTemplate")
		group.GET("findTemplate", apiTemplate.FindTemplate)
		group.GET("getTemplateList", apiTemplate.GetTemplateList)
	}
}
